x/oracle/keeper/oracle.go

```diff
@@ -60,6 +66,34 @@ func (k Keeper) CheckValidatorStatus(ctx sdk.Context, oracleAccAddr sdk.AccAddre
 	return nil
 }
 
+// checkOracleRegistrationStatus checks the status of OracleRegistration and Oracle.
+func (k Keeper) checkOracleRegistrationStatus(ctx sdk.Context, uniqueID, oracleAddress string) error {
+	existing, err := k.GetOracleRegistration(ctx, uniqueID, oracleAddress)
+	if err != nil {
+		if errors.Is(err, types.ErrOracleRegistrationNotFound) {
+			return nil
+		}
+		return err
+	}
+
+	if existing.Status == types.ORACLE_REGISTRATION_STATUS_VOTING_PERIOD {
+		return fmt.Errorf("in voting period")
+	}
+
+	if existing.Status == types.ORACLE_REGISTRATION_STATUS_PASSED {
+		oracle, err := k.GetOracle(ctx, oracleAddress)
+		if err != nil {
+			return err
+		}
+
+		if oracle.Status != types.ORACLE_STATUS_JAILED {
+			return fmt.Errorf("only jailed oracle can re-register oracle")
+		}
+	}
+
+	return nil
```

Comment:
How about handling for all states like this?
In this case, if a new state is added later, an error will be raised and unexpected behavior can be detected.

```suggestion
	switch existing.Status {
	case types.ORACLE_REGISTRATION_STATUS_PASSED:
		oracle, err := k.GetOracle(ctx, oracleAddress)
		if err != nil {
			return err
		}

		if oracle.Status != types.ORACLE_STATUS_JAILED {
			return fmt.Errorf("only jailed oracle can re-register oracle")
		}

		return nil
	case types.ORACLE_REGISTRATION_STATUS_VOTING_PERIOD:
		return fmt.Errorf("in voting period")
	case types.ORACLE_REGISTRATION_STATUS_REJECTED:
		return nil
	default:
		return fmt.Errorf("unexpected state. status: %s", existing.Status)
	}
```

Reply:
Thank you.
As you said, it would be better to detect unexpected status. I'll improve it as you suggested.